framework/infrastructure/gcp/api.py

```diff
@@ -96,6 +112,11 @@ def __init__(
 
     def close(self):
         self._exit_stack.close()
+        # Reset value of global googleapiclient.model.dump_request_response,
+        # but only if previously changed it.
+        if self._dump_req_resp is not None:
```

Comment:
[Minor] Can be simplified to `if self._dump_req_resp:`

Reply:
Not really, we need to distinguish here between `False` and `None`. In our, the previous value is is `False`, so the global will never be reset if we do if `self._dump_req_resp`.